solr/solrj/src/java/org/apache/solr/common/cloud/Slice.java

```diff
@@ -183,6 +185,12 @@ public Slice(String name, Map<String,Replica> replicas, Map<String,Object> props
     this.replicas = replicas != null ? replicas : makeReplicas(collection,name, (Map<String,Object>)propMap.get(REPLICAS));
     propMap.put(REPLICAS, this.replicas);
 
+    this.replicas.values().forEach(r -> {
```

Comment:
Really minor! Just wondering if it's slightly simpler if u a simple long `numLeaderReplicas` with:
`this.replicas.values().stream().filter(r -> Replica.Type.isLeaderType(r.getType())).count();` ? 🤔

Reply:
Good call, that makes it so we can just have an int too instead of an AtomicInteger